### PR TITLE
Major refactor of clinvar_raw to simplify code flow

### PR DESCRIPTION
**Pull request by theferrit32**
- Removes use of core.async channels and futures
- Converts some functions into pure functions with no side effects

### CODE REVIEW

**Comment by larrybabb** [comment]
@theferrit32 is there an issue on the github board to associate with this?

**Comment by theferrit32** [comment]
@larrybabb this was done as part of the work to ingest the larger test dataset. It significantly speeds things up, and makes it less error prone. I would like to clean this up before merging to master but it is not a high priority because it is not blocking anything else right now.

An issue could be added and put in the backlog.

File: src/clinvar_raw/core.clj
Comment (tbl3rd) [inline src/clinvar_raw/core.clj]: Why not just delete this?

File: src/clinvar_raw/stream.clj
Comment (tbl3rd) [inline src/clinvar_raw/stream.clj:57]: The docstring could explain bucket and filename.

File: src/clinvar_raw/stream.clj
Comment (tbl3rd) [inline src/clinvar_raw/stream.clj:71]: Document the parameters?

File: src/clinvar_raw/stream.clj
Comment (tbl3rd) [inline src/clinvar_raw/stream.clj]: Where is this used?

File: src/clinvar_raw/stream.clj
Comment (tbl3rd) [inline src/clinvar_raw/stream.clj]:
```clojure
(defn dissoc-nil-values
  "Remove entries with nil values from `input` map."
  [input]
  (into {} (remove (comp nil? val) input)))
```

File: src/clinvar_raw/stream.clj
Comment (tbl3rd) [inline src/clinvar_raw/stream.clj]: I left some comments on the /tbl/async-refactor/ branch.

**Review by larrybabb — APPROVED** [review]
As usual I will defer to the clojure experts for final say, but these refinements look sound to me. Pls don't delay in merging them if they are needed to get our urgent goals accomplished.

**Review by larrybabb — APPROVED** [review]

**Comment by theferrit32** [comment]
I've addressed feedback from Tom aside from the file processing. That should be revisited later to more match other patterns for large sequence or file processing in our code.
https://github.com/clingen-data-model/clinvar-streams/issues/63

**Pull request merged**
